src/main/java/duke/ChatBot.java

```diff
@@ -0,0 +1,170 @@
+     */
+    public String handleDone(int index) {
+        Task t = tasklist.completeTask(index);
+        s.saveTasks(tasklist);
+        return ui.showCompleteTaskMessage(t);
+    }
+
+    /**
+     * This method is to add a new deadline to the storage and
+     * print out the add message when the task is added successfully.
+     *
+     * @param name Name of the deadline.
+     * @param deadline The date and time of the deadline.
+     */
+    public String handleDeadline(String name, LocalDateTime deadline) {
+        Task t = tasklist.addDeadline(name, deadline, false);
+        s.saveTasks(tasklist);
+        return ui.showAddTaskMessage(t, tasklist.getTotalTasksNumber());
+    }
+
+    /**
+     * This method is to add a new task to the storage and
+     * print out the add message when the task is added successfully.
+     * @param name Name of the task.
+     */
+    public String handleTodo(String name) {
+        Task t = tasklist.addTodo(name, false);
+        s.saveTasks(tasklist);
+        return ui.showAddTaskMessage(t, tasklist.getTotalTasksNumber());
+    }
+
+    /**
+     * This method is to add a new event to the storage and
+     * print out the add message when the task is added successfully.
+     *
+     * @param name Name of the deadline.
+     * @param time The date and time of the event.
+     */
+    public String handleEvent(String name, LocalDateTime time) {
+        Task t = tasklist.addEvent(name, time, false);
+        s.saveTasks(tasklist);
+        return ui.showAddTaskMessage(t, tasklist.getTotalTasksNumber());
+
+    }
+
+    /**
+     * This method deletes the task specified by the user and
+     * print the delete message on successful deletion.
+     *
+     * @param index Index of the task specified by the user.
+     */
+    public String handleDelete(int index) {
+        Task t = tasklist.deleteTask(index);
+        s.saveTasks(tasklist);
+        return ui.showDeleteTaskMessage(t, tasklist.getTotalTasksNumber());
+    }
+
+    /**
+     * This method calls TaskList to find tasks related to the
+     * input by the user and calls UI to display them.
+     * @param name User input task to find.
+     */
+    public String handleFind(String name) {
+        String findTask = tasklist.findTask(name);
+        return ui.showFoundTask(findTask);
+    }
+
+    /**
+     * This method prints the list of all commands.
+     */
+    public String handleWrongCommand() {
+        return ui.showListOfCommands();
+    }
+
+    /**
+     * This method prints the error message thrown by the program.
+     * @param message The error message.
+     */
+    public String handleErrorMessage(String message) {
+        return ui.showErrorMessage(message);
+    }
+
+    /**
+     * Calls the UI to return the start message.
+     */
+    public String handleStart() {
+        return ui.showStartMessage();
+    }
+
+    /**
+     * This method gets the total number of tasks.
+     * @return total number of tasks.
```

Comment:
I think there should be a blank line between the description and the param/return

Reply:
Noted, will make the changes